010-provider-architecture/002-final/lib/ui/views/home_view.dart

```diff
@@ -8,46 +8,75 @@ import 'package:provider_architecutre/ui/shared/app_colors.dart';
 import 'package:provider_architecutre/ui/shared/text_styles.dart';
 import 'package:provider_architecutre/ui/shared/ui_helpers.dart';
 import 'package:provider_architecutre/ui/widgets/postlist_item.dart';
+import 'package:functional_widget_annotation/functional_widget_annotation.dart';
 
-import 'base_view.dart';
+part 'home_view.g.dart';
 
-class HomeView extends StatelessWidget {
-  @override
-  Widget build(BuildContext context) {
-    return BaseView<HomeModel>(
-      onModelReady: (model) => model.getPosts(Provider.of<User>(context).id),
-      builder: (context, model, child) => Scaffold(
-        backgroundColor: backgroundColor,
-        body: model.state == ViewState.Busy
-        ? Center(child: CircularProgressIndicator())
-         : Column(
-           crossAxisAlignment: CrossAxisAlignment.start,
-           children: <Widget>[
-             UIHelper.verticalSpaceLarge(),
-             Padding(
-              padding: const EdgeInsets.only(left: 20.0),
-              child: Text('Welcome ${Provider.of<User>(context).name}', 
-              style: headerStyle,),
-            ),
-            Padding(
-              padding: const EdgeInsets.only(left: 20.0),
-              child: Text('Here are all your posts', 
-              style: subHeaderStyle),
-            ),
-            UIHelper.verticalSpaceSmall(),
-            Expanded(child: getPostsUi(model.posts)),
-        ],)
+@widget
+Widget homeView() {
+  return Scaffold(
+    backgroundColor: backgroundColor,
+    // A consumer so that we don't rebuild `Scaffold` when `HomeModel` change.
+    body: Consumer<HomeModel>(builder: (context, model, _) {
+      return model.state == ViewState.Busy
+          ? const Center(child: CircularProgressIndicator())
+          : const HomeContent(
+              header: HeaderTitle(),
+              body: PostList(),
+            );
+    }),
+  );
+}
+
+// It's better as a standalone widget since it depends on `BuiltContext`
+// so we don't rebuild `HomeContent` for no reason, and we have a const constructor
+@widget
```

Comment:
This is pretty sick. I'll be using this and writing a tutorial on it as well. Thanks for showing me this trick